src/Orchard.Web/Modules/Orchard.AntiSpam/Drivers/ReCaptchaPartDriver.cs

```diff
@@ -67,67 +70,55 @@ public class ReCaptchaPartDriver : ContentPartDriver<ReCaptchaPart> {
                 return null;
             }
 
-            var submitViewModel = new ReCaptchaPartSubmitViewModel();
-
-            if(updater.TryUpdateModel(submitViewModel, String.Empty, null, null)) {
-                var context = workContext.HttpContext;
-
-                try {
-                    var result = ExecuteValidateRequest(
-                        settings.PrivateKey,
-                        context.Request.ServerVariables["REMOTE_ADDR"],
-                        submitViewModel.recaptcha_challenge_field,
-                        submitViewModel.recaptcha_response_field
-                        );
-
-                    if (!HandleValidateResponse(context, result)) {
-	                _notifier.Error(T("The text you entered in the Captcha field does not match the image"));
-                    	updater.AddModelError("", T("The text you entered in the Captcha field does not match the image"));
+            var context = workContext.HttpContext;
+
+            try {
+                var result = ExecuteValidateRequest(
+                    settings.PrivateKey,
+                    context.Request.ServerVariables["REMOTE_ADDR"],
+                    context.Request.Form["g-recaptcha-response"]
+                    );
+
+                ReCaptchaPartResponseModel responseModel = _jsonConverter.Deserialize<ReCaptchaPartResponseModel>(result);
+
+                if (!responseModel.Success) {

```

Comment:
I was wondering about that but changed it to a string since that's what was
requested. Should I change it back to a boolean or change the if statement?
Do I need to make the change or will you guys since it's been merged?

On Fri, Jul 24, 2015, 4:15 AM Sipke Schoorstra <notifications@github.com>
wrote:

> In src/Orchard.Web/Modules/Orchard.AntiSpam/Drivers/ReCaptchaPartDriver.cs
> <https://github.com/OrchardCMS/Orchard/pull/5332#discussion_r35411018>:
>
> > -
> > -                    if (!HandleValidateResponse(context, result)) {
> > -	                _notifier.Error(T("The text you entered in the Captcha field does not match the image"));
> > -                    	updater.AddModelError("", T("The text you entered in the Captcha field does not match the image"));
> > +            var context = workContext.HttpContext;
> > +
> > +            try {
> > +                var result = ExecuteValidateRequest(
> > +                    settings.PrivateKey,
> > +                    context.Request.ServerVariables["REMOTE_ADDR"],
> > +                    context.Request.Form["g-
>
> recaptcha-response"]
> > +                    );
> > +
> > +                ReCaptchaPartResponseModel responseModel = _jsonConverter.Deserialize<ReCaptchaPartResponseModel>(result);
> > +
> > +                if (!responseModel.Success) {
>
> This line doesn't compile since Success is a string, not a boolean.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/OrchardCMS/Orchard/pull/5332/files#r35411018>.
>